cmake/CMakeLists.txt

```diff
@@ -1005,12 +1010,14 @@ function(onnxruntime_set_compile_flags target_name)
       # Enable warning
       target_compile_options(${target_name} PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:--compiler-options -Wall>" "$<$<NOT:$<COMPILE_LANGUAGE:CUDA>>:-Wall>")
       target_compile_options(${target_name} PRIVATE "$<$<NOT:$<COMPILE_LANGUAGE:CUDA>>:-Wextra>")
-
       if(onnxruntime_DEV_MODE)
           target_compile_options(${target_name} PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:--compiler-options -Werror>" "$<$<NOT:$<COMPILE_LANGUAGE:CUDA>>:-Werror>")
       endif()
 
       target_compile_definitions(${target_name} PUBLIC -DNSYNC_ATOMIC_CPP11)
+      if(APPLE)
+        target_compile_definitions(${target_name} PRIVATE -Doptional_CONFIG_SELECT_OPTIONAL=1)
```

Comment:
we can use the value `optional_OPTIONAL_NONSTD`, according to the docs: https://github.com/martinmoene/optional-lite#select-stdoptional-or-nonstdoptional

also, is ORT's optional.h a public header? include/onnxruntime/core/common/optional.h
if it may be included by external code, that code would need to configure it the same way. should it be a PUBLIC compile definition?

Reply:
should it be PUBLIC instead of PRIVATE?

Reply:
Done.